client-app/src/Components/StatusUpdate.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState, ChangeEvent, FormEvent } from 'react';
+import axios from 'axios';
+import { useParams } from 'react-router-dom';
+import { useNavigate } from 'react-router-dom';
+import ItemStatus from '../constants/Enums';
+import '../css/DonorForm.css';
+
+interface FormData {
+    currentStatus: string;
+    dateDonated: string;
+}
+
+interface FormErrors {
+    [key: string]: string;
+}
+
+interface Option {
+    value: string;
+    label: string;
+    id?: number;
+}
+
+const StatusUpdate: React.FC = () => {
+    const navigate = useNavigate();
+    const [formData, setFormData] = useState<FormData>({
+        currentStatus: 'Line 25',  // Initial status
+        dateDonated: '',
+    });
+    const { id } = useParams<{ id: string }>();
+    const [errors, setErrors] = useState<FormErrors>({});
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
+    const [successMessage, setSuccessMessage] = useState<string | null>(null);
+
+    const handleChange = async (e: ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
+        const { name, value } = e.target;
+        setFormData(prevState => ({
+            ...prevState,
+            [name]: value,
+        }));
+        setErrors(prevState => ({ ...prevState, [name]: '' }));
+        setErrorMessage(null);
+        setSuccessMessage(null);
+    };
+
+    const validateField = (name: string, value: any) => {
+        if (!value || value.length === 0) {
+            return `${name.replace(/([A-Z])/g, ' $1')} is required`;
+        }
+        return '';
+    };
+
+    const validateForm = (): boolean => {
+        const newErrors: FormErrors = {};
+        Object.keys(formData).forEach(field => {
+            const error = validateField(field, (formData as any)[field]);
+            if (error) newErrors[field] = error;
+        });
+        setErrors(newErrors);
+        return Object.keys(newErrors).length === 0;
+    };
```

Comment:
Validation failed even when after giving non null values for both the fields. Can you please fix that?